src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/sampler/HTTPSamplerBase.java

```diff
@@ -297,7 +297,7 @@
     protected static final int MAX_FRAME_DEPTH = JMeterUtils.getPropDefault("httpsampler.max_frame_depth", 5); // $NON-NLS-1$
 
     // Derive the mapping of content types to parsers
-    private static final Map<String, String> PARSERS_FOR_CONTENT_TYPE = new HashMap<>();
+    private static final Map<String, String> PARSERS_FOR_CONTENT_TYPE = new ConcurrentHashMap<>();
```

Comment:
Was it a bug here ? If so isn't it worth creating a bugzilla ?

Reply:
No.
The thing is the map was initialized from `jmeter.properties` in `static {...}` initializer.

That does not play well for testing, as I don't really have `jmeter.properties` in a unit kind of test.

That is why I exposed a `package-private` method `registerParser`, and thus I made the map to be `concurrent` just in case.

Now I realize I might need to clean up the contents of the map `after the test`, however, I don't really like how the configuration is parsed in the static block. I would like to rework that to constructor parameter or setter injection, so one does not need filesystem (and files like `jmeter.properties`) in order to test `HTTPSamplerBase` class.